### PR TITLE
HPCC-32397 Ensure delta writer is stopped earlier to avoid crash

### PR DESCRIPTION
**Pull request by jakesmith**
The delta writer must be stopped during closedown, before other dependent members of the SDS manager are destroyed.

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/base/dasds.cpp [hunk 1]
```diff
@@ -2225,7 +2225,7 @@ void CDeltaWriter::addToQueue(CTransactionItem *item)
         ++totalQueueLimitHits;
         // force a synchronous save
         CCycleTimer timer;
-        PROGLOG("Forcing synchronous save of %u transactions", (unsigned)pending.size());
+        PROGLOG("Forcing synchronous save of %u transactions (pendingSz=%zu)", (unsigned)pending.size(), pendingSz);
```
Comment (jakesmith) [inline dali/base/dasds.cpp:2228]: totally incidental change - feel free to reject for it to go into a separate PR

File: dali/base/dasds.cpp [hunk 2]
```diff
@@ -6056,6 +6056,7 @@ CCovenSDSManager::~CCovenSDSManager()
 {
     if (unhandledThread) unhandledThread->join();
     if (coalesce) coalesce->stop();
+    deltaWriter.stop();
```
Comment (jakesmith) [inline dali/base/dasds.cpp:6059]: NB: before this change, stop() would happen via member dtor.

**Review by jakesmith — COMMENTED** [review]
@ghalliday - please review.

Wasn't sure whether to target 9.6 or 9.8 - decided on 9.8 and pondered whether should go to master, because it is long term bug, that affects shutdown only.

**Comment by ghalliday** [comment]
I think it is fair to merge fixes for crashes and cores into the current release as long as they are relatively safe

**Review by ghalliday — APPROVED** [review]

**Pull request merged**